code/modules/mob/living/carbon/xenomorph/mutators/strains/drone/gardener.dm

```diff
@@ -358,3 +361,44 @@
 	set hidden = 1
 	var/action_name = "Plant Hardy Weeds (125)"
 	handle_xeno_macro(src, action_name)
+
+/datum/behavior_delegate/drone_gardener
+	name = "Gardener Drone Behavior Delegate"
+
+	var/fruit_on_cooldown = FALSE
+	var/mutable_appearance/fruit_sac_overlay_icon
+
+/datum/behavior_delegate/drone_gardener/add_to_xeno()
+	on_update_icons()
+
+/datum/behavior_delegate/drone_gardener/on_update_icons()
+	if(!fruit_sac_overlay_icon)
+		fruit_sac_overlay_icon = mutable_appearance('icons/mob/xenos/drone_strain_overlays.dmi',"Gardener Drone Walking")
+
+	bound_xeno.overlays -= fruit_sac_overlay_icon
+	fruit_sac_overlay_icon.overlays.Cut()
+
+	//var/image/image_to_add
+	if(bound_xeno.stat == DEAD)
+		fruit_sac_overlay_icon.icon_state = "Gardener Drone Knocked Down[fruit_on_cooldown ? "_spent" : ""]"
+	else if(bound_xeno.lying)
+		if((bound_xeno.resting || bound_xeno.sleeping) && (!bound_xeno.knocked_down && !bound_xeno.knocked_out && bound_xeno.health > 0))
+			fruit_sac_overlay_icon.icon_state = "Gardener Drone Sleeping[fruit_on_cooldown ? "_spent" : ""]"
+		else
+			fruit_sac_overlay_icon.icon_state = "Gardener Drone Knocked Down[fruit_on_cooldown ? "_spent" : ""]"
+	else
+		fruit_sac_overlay_icon.icon_state = "Gardener Drone Walking[fruit_on_cooldown ? "_spent" : ""]"
+
```

Comment:
or you could just do if fruit on cooldown icon_state = "[icon_state]_spent"

Reply:
wtf!